### PR TITLE
feat: change address to major

### LINKED ISSUE #27: Update Address Book to Contact Book
Update "Address Book" to "Contact Book" in our [User Guide](https://ay2324s2-cs2103t-f13-1.github.io/tp/UserGuide.html) and [Developer Guide](https://ay2324s2-cs2103t-f13-1.github.io/tp/DeveloperGuide.html) to be more align with our product TA Pro.

### LINKED ISSUE #107: URGENT - Change "Address" to "Major"
Non exhaustively, changes to be done in:
1. Prefix
2. Documentation (DG and UG)
3. UML diagrams (if have mention of address)
4. Class name
5. Default messages when no major is given
6. Default json when data file is not present

### PR DESCRIPTION
**Pull request by tituschewxj**
Fix
* Inappropriate changes from address to major

Add
* Parameter syntax for major

Remove
* The three tutorial files from our docs, which are not that useful. (which is why a lot of lines removed)

>[!NOTE]
>
>We need to clear the existing data as we introduce a new field, otherwise the contact book is cleared of data.

This PR is based off #141

Resolves #107, (somehow also) resolves #27

### CODE REVIEW

File: docs/team/xiaoyun.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ title: "Xiaoyun's Project Portfolio Page"
 
 ### Project: AddressBook Level 3
 
-AddressBook - Level 3 is a desktop address book application used for teaching Software Engineering principles. The user interacts with it using a CLI, and it has a GUI created with JavaFX. It is written in Java, and has about 10 kLoC.
+AddressBook - Level 3 is a desktop contact book application used for teaching Software Engineering principles. The user interacts with it using a CLI, and it has a GUI created with JavaFX. It is written in Java, and has about 10 kLoC.
```
Comment (belligerentbeagle) [inline docs/team/xiaoyun.md:8]: ditto until this part

**Review by belligerentbeagle — APPROVED** [review]
Besides the comments left above, the rest LGTM! Thanks so much for being my second pair of eyes and for helping out with the fixes. Really a big and great combined effort!

File: docs/team/anant.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ title: "Anant's Project Portfolio Page"
 
 ### Project: AddressBook Level 3
 
-AddressBook - Level 3 is a desktop address book application used for teaching Software Engineering principles. The user interacts with it using a CLI, and it has a GUI created with JavaFX. It is written in Java, and has about 10 kLoC.
+AddressBook - Level 3 is a desktop contact book application used for teaching Software Engineering principles. The user interacts with it using a CLI, and it has a GUI created with JavaFX. It is written in Java, and has about 10 kLoC.
```
[the remark this answers is not in the document]
Comment (tituschewxj) [reply docs/team/anant.md]: Can, but if we want we can update our own page later.

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 3]
```diff
@@ -111,16 +111,21 @@ public void parseAddress_null_throwsNullPointerException() {
...
     }
 
     @Test
-    public void parseAddress_validValueWithWhitespace_returnsTrimmedAddress() throws Exception {
-        String addressWithWhitespace = WHITESPACE + VALID_ADDRESS + WHITESPACE;
-        Address expectedAddress = new Address(VALID_ADDRESS);
-        assertEquals(expectedAddress, ParserUtil.parseAddress(addressWithWhitespace));
+    public void parseMajor_validValueWithoutWhitespace_returnsMajor() throws Exception {
+        Major expectedMajor = new Major(VALID_MAJOR);
+        assertEquals(expectedMajor, ParserUtil.parseAddress(VALID_MAJOR));
```
Comment (Anant1902) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java:121]: I think this method can be changed to parseMajor instead?
```suggestion
        assertEquals(expectedMajor, ParserUtil.parseMajor(VALID_MAJOR));
```

**Review by Anant1902 — CHANGES_REQUESTED** [review]
Just 1 tiny nit! Thanks for grinding this out Titus!

**Review by Anant1902 — APPROVED** [review]
LGTM! Thanks for the change!

**Pull request merged**